tel2puml/find_unique_graphs/otel_ingestion/otel_data_holder.py

```diff
@@ -12,9 +21,133 @@ class DataHolder(ABC):
 
     @abstractmethod
     def save_data(self, otel_event: OTelEvent) -> None:
-        """Method for batching and saving OTel data.
+        """Abstract method for batching and saving OTel data.
 
         :param otel_event: An OTelEvent object
         :type otel_event: :class: `OTelEvent`
         """
         pass
+
+    @abstractmethod
+    def clean_up(self) -> None:
+        """Abstract method for any clean up tasks."""
+
+        pass
+
+
+class SQLDataHolder(DataHolder):
+    """A class to handle saving data in SQL databases using SQLAlchemy."""
+
+    def __init__(self, config: SQLDataHolderConfig) -> None:
+        """Constructor method.
+
+        :param config: Configuration parameters.
+        :type config: :class: `SQLDataHolderConfig`
+        """
+
+        self.node_models_to_save: list[NodeModel] = []
+        self.node_relationships_to_save: list[dict[str, str]] = []
+        self.batch_size: int = config["batch_size"]
+        self.engine: Engine = create_engine(config["db_uri"], echo=False)
+        self.session: Session = Session(bind=self.engine)
+        self.base: Base = Base()
+        self.create_db_tables()
+
+    def create_db_tables(self) -> None:
+        """Method to create the database tables based on the defined models."""
+
+        self.base.metadata.create_all(self.engine)
+
+    def save_data(self, otel_event: OTelEvent) -> None:
```

Comment:
in this can we set some attributes for max and min datetime so when we come to making the function to get the time window its much simpler